include/swift/SIL/SILInstruction.h

```diff
@@ -459,8 +463,15 @@ class UnaryInstructionWithTypeDependentOperandsBase :
     }
   }
 
+  // Work around MSVC bug: can't infer llvm::trailing_objects_internal,
+  // even though we granted friend access to it.
   size_t numTrailingObjects(
+#if defined(_MSC_VER) && !defined(__clang__)
+      llvm::trailing_objects_internal::TrailingObjectsBase::OverloadToken<
+      Operand>) const {
+#else
       typename TrailingBase::template OverloadToken<Operand>) const {
+#endif
```

Comment:
I think it would be better to consolidate the workaround in a single typedef somewhere, without the conditional.

```
// Equivalent to `typename TrailingBase::template OverloadToken<Operand>`.
// We have to fully elaborate the type because of an MSVC bug.
using OperandOverloadToken 
  = llvm::trailing_objects_internal::TrailingObjectsBase::OverloadToken<Operand>;
```